site/pages/api/pledge/index.ts

```diff
@@ -8,16 +8,28 @@ export default async function handler(
   switch (req.method) {
     case "PUT":
       try {
-        const sessionToken = req.headers.authorization?.split(" ")[1];
-        const data = await findOrCreatePledge({
-          pledge: req.body,
-          sessionToken,
+        const signature = req.headers.authorization?.split(" ")[1];
+
+        if (!signature) {
+          return res.status(400).json({ message: "Bad request" });
+        }
+
+        const pledge = await findOrCreatePledge({
+          pageAddress: req.body.pageAddress,
+          pledge: req.body.pledge,
+          signature,
         });
-        const pledge = JSON.parse(JSON.stringify(data));
 
-        res.status(200).json({ pledge: pledgeResolver(pledge) });
```

Comment:
don't you still need to return a res(200) to indicate success?

Reply:
hehe the diffs are misleading, it's been moved down a couple lines

https://github.com/KlimaDAO/klimadao/blob/5db1b812f4bc2627efb5940990957ef1791c307b/site/pages/api/pledge/index.ts#L29